operations/service_deploy_smoke_utils.go

```diff
@@ -216,43 +267,49 @@ func checkTaskLog(body []byte) error {
 		return errors.New("did not find task completed message in log")
 	}
 
-	// Validate that setup_group only runs in first task
-	if strings.Contains(page, "first") {
-		if !strings.Contains(page, "setup_group") {
-			return errors.New("did not find setup_group in logs for first task")
-		}
-	} else {
-		if strings.Contains(page, "setup_group") {
-			return errors.New("setup_group should only run in first task")
+	if mode == agent.HostMode {
+		// Validate that setup_group only runs in first task
+		if strings.Contains(page, "first") {
+			if !strings.Contains(page, "setup_group") {
+				return errors.New("did not find setup_group in logs for first task")
+			}
+		} else {
+			if strings.Contains(page, "setup_group") {
+				return errors.New("setup_group should only run in first task")
+			}
 		}
-	}
 
-	// Validate that setup_task and teardown_task run for all tasks
-	if !strings.Contains(page, "setup_task") {
-		return errors.New("did not find setup_task in logs")
-	}
-	if !strings.Contains(page, "teardown_task") {
-		return errors.New("did not find teardown_task in logs")
-	}
+		// Validate that setup_task and teardown_task run for all tasks
+		if !strings.Contains(page, "setup_task") {
+			return errors.New("did not find setup_task in logs")
+		}
+		if !strings.Contains(page, "teardown_task") {
+			return errors.New("did not find teardown_task in logs")
+		}
 
-	// Validate that teardown_group only runs in last task
-	if strings.Contains(page, "fourth") {
-		if !strings.Contains(page, "teardown_group") {
-			return errors.New("did not find teardown_group in logs for last (fourth) task")
+		// Validate that teardown_group only runs in last task
+		if strings.Contains(page, "fourth") {
+			if !strings.Contains(page, "teardown_group") {
+				return errors.New("did not find teardown_group in logs for last (fourth) task")
+			}
+		} else {
+			if strings.Contains(page, "teardown_group") {
+				return errors.New("teardown_group should only run in last (fourth) task")
+			}
 		}
-	} else {
-		if strings.Contains(page, "teardown_group") {
-			return errors.New("teardown_group should only run in last (fourth) task")
+	} else if mode == agent.PodMode {
+		if !strings.Contains(page, "container task") {
```

Comment:
Can we add a TODO here for PM-2617 to add task groups to the smoke test once they're supported?